### PR TITLE
KAFKA-4059: Documentation still refers to AsyncProducer and SyncProducer

### PR DESCRIPTION
**Pull request by tombentley**
Also remove old code snippet which bears little resemblance to the current
Producer API.

The contribution is my original work and I license the work to the project under the project's open source license.

### CODE REVIEW

File: docs/implementation.html [hunk 1]
```diff
@@ -20,26 +20,7 @@
```
Comment (ijuma) [inline docs/implementation.html]: I think we should remove the whole API design section.

**Review by ijuma — COMMENTED** [review]
Thanks for the PR. I left one comment.

File: docs/implementation.html (hunk 1 above)
Comment (tombentley) [reply docs/implementation.html]: OK, fair enough

**Comment by ijuma** [comment]
Thanks for the update, `toc.html` also needs to be updated.

**Comment by tombentley** [comment]
Ah, thanks! I'm still finding my way around the docs.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/5508/
Test PASSed (JDK 7 and Scala 2.11).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/5493/
Test PASSed (JDK 8 and Scala 2.12).

**Review by ijuma — APPROVED** [review]
Thanks for the PR, LGTM.

**Pull request closed without merging**